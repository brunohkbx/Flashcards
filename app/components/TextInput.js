import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { TextInput as _TextInput } from 'react-native-paper';
import ErrorText from './ErrorText';

const TextInput = props => {
  const {
    label,
    value,
    errorMessage,
    handleChangeText,
    handleBlur
  } = props;

  return (
    <View>
      <_TextInput
        label={label}
        value={value}
        onChangeText={handleChangeText}
        onBlur={handleBlur}
        error={errorMessage.length > 0}
        autoFocus
      />
      <ErrorText>{errorMessage}</ErrorText>
    </View>
  );
}

TextInput.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string,
  errorMessage: PropTypes.string,
  handleChangeText: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
}

TextInput.defaultProps = {
  value: '',
  errorMessage: ''
}

export default TextInput;
