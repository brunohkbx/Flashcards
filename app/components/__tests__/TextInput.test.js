import React from 'react';
import { shallow } from 'enzyme';
import TextInput from '../TextInput';

describe('TextInput', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      label: 'Title',
      handleChangeText: jest.fn(),
      handleBlur: jest.fn()
    }, propOverrides)

    const wrapper = shallow(<TextInput {...defaultProps} />);

    const blur = () => {
      wrapper.find('withTheme(TextInput)').simulate('blur');
    }

    const changeText = () => {
      wrapper.find('withTheme(TextInput)').simulate('changeText');
    }

    return {
      defaultProps,
      wrapper,
      blur,
      changeText
    }
  }

  test('onChangeText calls handleChangeText', () => {
    const mockHandleChangeText = jest.fn();
    const { changeText } = setup({ handleChangeText: mockHandleChangeText });

    changeText();

    expect(mockHandleChangeText).toHaveBeenCalled();
  });

  test('onBlur calls handleBlur', () => {
    const mockHandleBlur = jest.fn();
    const { blur } = setup({ handleBlur: mockHandleBlur });

    blur();

    expect(mockHandleBlur).toHaveBeenCalled();
  });

  it('highlights itself and show ErrorText if errorMessage is present', () => {
    const { wrapper } = setup({ errorMessage: 'foo' });

    expect(wrapper).toMatchSnapshot();
  });

  it('does not highlights itself and show ErrorText if errorMessage is not present', () => {
    const { wrapper } = setup({ errorMessage: undefined });

    expect(wrapper).toMatchSnapshot();
  });
});
