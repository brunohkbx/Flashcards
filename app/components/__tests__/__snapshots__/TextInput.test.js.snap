// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput does not highlights itself and show ErrorText if errorMessage is not present 1`] = `
<Component>
  <withTheme(TextInput)
    autoFocus={true}
    error={false}
    label="Title"
    onBlur={[MockFunction]}
    onChangeText={[MockFunction]}
    value=""
  />
  <Styled(Paragraph) />
</Component>
`;

exports[`TextInput highlights itself and show ErrorText if errorMessage is present 1`] = `
<Component>
  <withTheme(TextInput)
    autoFocus={true}
    error={true}
    label="Title"
    onBlur={[MockFunction]}
    onChangeText={[MockFunction]}
    value=""
  />
  <Styled(Paragraph)>
    foo
  </Styled(Paragraph)>
</Component>
`;
